prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -637,15 +637,29 @@ void rxr_pkt_handle_atomrsp_recv(struct rxr_ep *ep,
 	struct rxr_atomrsp_pkt *atomrsp_pkt = NULL;
 	struct rxr_atomrsp_hdr *atomrsp_hdr = NULL;
 	struct rxr_op_entry *tx_entry = NULL;
+	struct efa_mr *result_mr;
+	uint64_t device;
+	enum fi_hmem_iface hmem_iface;
+	int i;
 
 	atomrsp_pkt = (struct rxr_atomrsp_pkt *)pkt_entry->pkt;
 	atomrsp_hdr = &atomrsp_pkt->hdr;
 	tx_entry = ofi_bufpool_get_ibuf(ep->op_entry_pool, atomrsp_hdr->recv_id);
 
-	ofi_copy_to_iov(tx_entry->atomic_ex.resp_iov,
-			tx_entry->atomic_ex.resp_iov_count,
-			0, atomrsp_pkt->data,
-			atomrsp_hdr->seg_length);
+	for (i = 0; i < tx_entry->atomic_ex.resp_iov_count; i++) {
+		result_mr = tx_entry->atomic_ex.result_desc[i];
+		if (result_mr) {
+			hmem_iface = result_mr->peer.iface;
+			device = result_mr->peer.device.reserved;
+		} else {
+			hmem_iface = FI_HMEM_SYSTEM;
+			device = 0;
+		}
+
+		ofi_copy_to_hmem_iov(hmem_iface, device, tx_entry->atomic_ex.resp_iov,
```

Comment:
can we wrap this up into a utility function

Reply:
done

Reply:
done